vignettes/vignette_aspect_importance.Rmd

```diff
@@ -0,0 +1,203 @@
+---
+title: "Aspect importance function examples"
+author: "Katarzyna Pękala"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_document
+vignette: >
+  %\VignetteIndexEntry{Aspect importance function examples}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options: 
+  chunk_output_type: console
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = FALSE,
+  comment = "#>",
+  warning = FALSE,
+  message = FALSE
+)
+```
+
+
+# Data and logistic regression model for Titanic survival
+
+Vignette presents the aspect_importance() function on the datasets: `titanic_imputed` and `apartments` (both are available in the `DALEX` package). 
+At the beginning, we download `titanic_imputed` dataset and build logistic regression model.
+
+```{r read titanic data}
+library("DALEX")
+titanic <- titanic_imputed
+titanic$country <- NULL
+titanic_without_target <- titanic[,colnames(titanic)!="survived"]
+head(titanic)
+
+model_titanic_glm <-
+  glm(survived == "yes" ~ class + gender + age + sibsp + parch + fare + embarked,
+      titanic,
+      family = "binomial")
+
+```
+
+#  Preparing additional parameters
+
+Before using aspect_importance() we need to:
+
+* group features of the dataset into aspects, 
+* choose observation for which we want to explain aspects' importance.
+
+```{r build parameters for titanic model}
+aspects_titanic <-
+  list(
+    wealth = c("class", "fare"),
+    family = c("sibsp", "parch"),
+    personal = c("age", "gender"),
+    embarked = "embarked"
+  )
+passenger <- data.frame(
+  class = factor(
+    "3rd",
+    levels = c("1st", "2nd", "3rd", "deck crew", "engineering crew", "restaurant staff", "victualling crew")),
+  gender = factor("male", levels = c("female", "male")),
+  age = 8,
+  sibsp = 0,
+  parch = 0,
+  fare = 18,
+  embarked = factor(
+    "Southampton",
+    levels = c("Belfast", "Cherbourg", "Queenstown", "Southampton")
+  )
+)
+
+passenger
+predict(model_titanic_glm, passenger, type = "response")
+
+```
+
+# Calculating aspect importance (logistic regression)
+
+Now we can call aspect_importance() function and see that features included in `wealth` (that is `class` and `fare`) have the biggest contribution on survival prediction for the passenger. That contribution is of negative type. Rest of the aspects have significantly smaller influence. However, in case of `family` and `personal`, it's a positive type of influence.
+
+```{r Calculating aspect importance (logistic regression)}
+library("ggplot2")
+library("ingredients")
```

Comment:
I guess that we should change `library("ingredients")` to `library("DALEXtra")` in vignette and in test file. Not sure how to do that at this point in the process, hence the review comment.